ls1intum/Artemis
# Display plagiarism run details

### Checklist

### Motivation and Context
To make the inspection of plagiarism results more insightful, this PR adds a new tab to the plagiarism sidebar to display additional information (duration of the run, similarity distribution).

### Description
I added a new `PlagiarismRunDetails` component to display the duration and similarity distribution of the plagiarism detection run. It can be viewed by clicking on "Run details" in the sidebar. Details are available for text, programming, and modeling exercises.

### Steps for Testing
**Note**: You need an exercise with multiple, similar submissions to properly test this PR.

1. Log in to Artemis
2. Navigate to Course Administration
3. Go to Exercises of a course
4. Click 'View' on a modeling, text, or programming exercise
5. Click 'Check Plagiarism'
6. On the plagiarism page, click 'Check Plagiarism' again

Make sure that

* In the sidebar, you should see the new "Run details" tab
* Selecting "Run details" should open the run details view
* The duration should be displayed and formatted correctly (HH:mm:ss.SSS)
* The similarity distribution should be rendered as a bar chart
* The bar chart should display the similarities (compare with plagiarism matches in the sidebar)
* Details should be available for text, programming, and modeling exercises

**Note**: If you only have a few submissions for an exercise, the plagiarism detection might be very fast (<1ms). In this case, the duration will be displayed as 00:00:00.000. This is correct behavior; just make sure no errors are thrown in the browser console.

### Test Coverage
plagiarism-sidebar.component.ts **81.25%**
plagiarism-run-details.component.ts **93.75%**

### Screenshots
<img width="1680" alt="Screenshot 2021-01-07 at 16 20 29" src="https://user-images.githubusercontent.com/21973908/103909688-4b4d2c80-5104-11eb-8342-8d283a4706b1.png">
